--- imaplib2.py	2010-06-06 12:24:46.000000000 +0100
+++ imaplib2_modded.py	2011-01-25 22:21:36.653189828 +0000
@@ -1579,7 +1579,7 @@
             if self.state == LOGOUT:
                 timeout = 1
             else:
-                timeout = None
+                timeout = 660
             try:
                 r = poll.poll(timeout)
                 if __debug__: self._log(5, 'poll => %s' % `r`)
@@ -1638,7 +1638,7 @@
             if self.state == LOGOUT:
                 timeout = 1
             else:
-                timeout = None
+                timeout = 660
             try:
                 r,w,e = select.select([self.read_fd], [], [], timeout)
                 if __debug__: self._log(5, 'select => %s, %s, %s' % (r,w,e))
